packages/insomnia-inso/src/commands/run-tests.js

```diff
@@ -60,38 +86,31 @@ export async function runInsomniaTests(
     return false;
   }
 
-  // Find environment
-  const workspaceId = suites[0].parentId;
-  const environment = env
-    ? loadEnvironment(db, workspaceId, env)
-    : await promptEnvironment(db, !!ci, workspaceId);
+  // Find env
+  const environment = await getEnvironments(db, ci, suites, env);
 
   if (!environment) {
     logger.fatal('No environment identified; cannot run tests without a valid environment.');
     return false;
   }
 
   // Generate test file
-  const testFileContents = await generate(
-    suites.map(suite =>
-      createTestSuite(
-        suite,
-        db.UnitTest.filter(t => t.parentId === suite._id),
-      ),
-    ),
-  );
+  const testFiles = await getTestFileContent(db, suites);
```

Comment:
This actually generates a single test file

```suggestion
  const testFile = await getTestFile(db, suites);
```